### PR TITLE
Fix remaining `torch_dtype` deprecation warnings on load and save paths

### LINKED ISSUE #25021: [BUG] `torch_dtype` deprecation warning still emitted on explicit-dtype load and dict-input save
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Issues Policy acknowledgement

- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### MLflow version

- master (reproduced at commit c46e5031f, after #24929 was merged)

### System information

- macOS 15.6.1 (Apple Silicon), Python 3.11, transformers 5.14.1, torch 2.13.0

### Describe the problem

#24929 fixed the `torch_dtype is deprecated` warning reported in #24875 for the common case of loading a saved transformers model, but two paths still emit it with transformers >= 4.56:

1. Loading with an explicit dtype, e.g. `mlflow.transformers.load_model(uri, torch_dtype=...)`. The kwarg is read but not removed from `kwargs`, so `conf.update(**kwargs)` in `_load_model` re-adds the deprecated name and it reaches `transformers.pipeline()`.
2. Saving a components dict that contains a `torch_dtype` entry. `_build_pipeline_from_model_input` forwards the key to `pipeline()` unchanged.

A fix with regression tests is up at #25017.

### Code to reproduce issue

Run each path in a fresh Python process. transformers deduplicates the warning within a process (`logger.warning_once`), so a single script running both paths prints it only once.

Path 1: loading with an explicit dtype

```python
import torch
import transformers
import mlflow

pipe = transformers.pipeline(
    "text-classification", model="distilbert-base-uncased-finetuned-sst-2-english"
)
mlflow.transformers.save_model(transformers_model=pipe, path="model")
mlflow.transformers.load_model("model", torch_dtype=torch.float16)
# -> `torch_dtype` is deprecated! Use `dtype` instead!
```

Path 2: saving a components dict

```python
import torch
import transformers
import mlflow

pipe = transformers.pipeline(
    "text-classification", model="distilbert-base-uncased-finetuned-sst-2-english"
)
mlflow.transformers.save_model(
    transformers_model={
        "model": pipe.model,
        "tokenizer": pipe.tokenizer,
        "torch_dtype": torch.float16,
    },
    task="text-classification",
    path="model2",
)
# -> `torch_dtype` is deprecated! Use `dtype` instead!
```

### Stack trace

No exception. The warning is: `torch_dtype` is deprecated! Use `dtype` instead!

### What component(s) does this bug affect?

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow model serving, deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/uiux`: Front-end, user experience, plotting
- [ ] `area/docs`: MLflow documentation pages

### PR DESCRIPTION
**Pull request by rrioh**
### Related Issues/PRs

Closes #25021 (follow-up to #24929)

### What changes are proposed in this pull request?

#24929 stopped the `torch_dtype is deprecated` warning for the main model load path, but two paths still trigger it on transformers >= 4.56:

- loading with an explicit dtype (`mlflow.transformers.load_model(uri, torch_dtype=...)`): the raw kwarg survives into `conf.update(**kwargs)` and reaches `pipeline()` under its deprecated name
- saving a components dict (`save_model(transformers_model={..., "torch_dtype": ...})`): `_build_pipeline_from_model_input` forwards the key to `pipeline()` as-is

This PR closes both. The load path now pops both the `torch_dtype` and `dtype` spellings from the kwargs and forwards a single argument under the name the installed transformers version expects. It reuses `_get_torch_dtype_kwarg_name()` from #24929 instead of adding a second helper. When both spellings are passed, `dtype` wins, which matches how transformers itself resolves them. The save path renames the key on a copy of the input dict, so the caller's dict stays untouched.

The MLmodel metadata format and the public `save_model`/`load_model` APIs are unchanged.

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

Two new tests cover the two paths. Both fail on current master and pass with this change. The load test is parametrized over both spellings; the `dtype` spelling already worked on master, so that case passes there too. They spy on the kwargs that actually reach `pipeline()` and assert that only the version-appropriate spelling is present, and that no deprecation message shows up in the transformers logger output (the warning goes through `logger.warning_once`, which `pytest.warns` cannot catch). The expectations adapt to the installed transformers version, so the same tests exercise the old-name branch on pre-4.56 versions in the CI matrix.

I also ran the dtype-related test selection against transformers 5.14.1 and 4.55.4 locally: the tests for this change pass on both versions. On master with transformers 5.14.1, exactly the two new tests fail. The only local failure is a pre-existing float64 case that fails identically on master, because Apple Silicon's MPS backend has no float64 support.

### Does this PR require documentation update?

- [x] No.

### Does this PR require updating the [MLflow Skills](https://github.com/mlflow/skills) repository?

- [x] No.

#### Is this a user-facing change?

- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Fixed `torch_dtype` deprecation warnings that still appeared when loading a transformers model with an explicit dtype or saving one from a components dict.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes

#### Is this PR a critical bugfix or security fix that should go into the next patch release?

- [ ] This PR is critical and needs to be in the next patch release
- [x] This PR can wait for the next minor release

### CODE REVIEW

**Review by PattaraS — APPROVED** [review]
LGTM — this correctly closes the two remaining `torch_dtype` warning paths that #24929 didn't cover, and it reuses `_get_torch_dtype_kwarg_name()` so version detection stays single-sourced. The `kwargs.pop(...)` on the load path is the right fix: the raw name was previously re-injected by `conf.update(**kwargs)` after the renamed one was set.

I ran the three new tests locally on transformers 5.x:
- With the fix: 4 passed (the explicit-dtype test is parametrized ×2).
- Reverting only the source change and keeping the tests: `load[torch_dtype]` and `save_dict_input` fail with the literal `` `torch_dtype` is deprecated! Use `dtype` instead! `` in the captured logger output, while `load[dtype]` and the no-explicit-dtype test pass — matching your write-up.

Approving. A few non-blocking nits for a follow-up if you like:

1. **Precedence when both spellings are passed at once.** On the load path `dtype` wins (`dtype_arg or torch_dtype_arg`); on the save path `model_dict[name] = <popped torch_dtype>` overwrites an existing `dtype`, so `torch_dtype` wins. Only matters for a components dict carrying both keys, but aligning both to "dtype wins" would be tidier.
2. **Test couples to a transformers internal**: `transformers.utils.logging.warning_once.cache_clear()` assumes that stays an `lru_cache`. Fine today, just mild version fragility.
3. **Latent footgun**: the save-path test works only because `pipeline` is imported *inside* `_build_pipeline_from_model_input`. If that's ever hoisted to a module-level `from transformers import pipeline`, the `mock.patch("transformers.pipeline")` spy goes dead and the test silently stops covering the call — maybe worth a one-line comment on the test.
